### PR TITLE
Implements NLTK and Spacy-based TextSplitters

### PR DESCRIPTION
**Pull request by delip**
This PR is for Issue #88 

- [x] `make format`
- [x] `make lint`
- [x] `make tests`

### CODE REVIEW

File: langchain/text_splitter.py [hunk 1]
```diff
@@ -46,3 +37,72 @@ def split_text(self, text: str) -> List[str]:
...
+        chunk_size: int = 4000,
+        chunk_overlap: int = 200,
+    ):
+        """Initialize the spacy text splitter."""
+        super(SpacyTextSplitter, self).__init__(separator, chunk_size, chunk_overlap)
+        try:
+            import spacy
+        except ImportError:
+            raise ImportError(
+                "Spacy is not installed, please install it with " "`pip install spacy`."
```
Comment (hwchase17) [inline langchain/text_splitter.py:101]: nit: think the formatter did this, but this is two strings (...with " "`pip...) but can just be one

File: tests/unit_tests/test_text_splitter.py [hunk 2]
```diff
@@ -26,3 +30,38 @@ def test_character_text_splitting_args() -> None:
     """Test invalid arguments."""
     with pytest.raises(ValueError):
         CharacterTextSplitter(chunk_size=2, chunk_overlap=4)
+
+
+def test_nltk_text_splitting_args() -> None:
```
Comment (hwchase17) [inline tests/unit_tests/test_text_splitter.py:35]: since these involve external dependencies (nltk/spacy) can move them to the integration tests folder?

**Review by hwchase17 — COMMENTED** [review]
this is fantastic

can we add nltk/spacy to the requirements.txt file? in the same place as all the other integration packages https://github.com/hwchase17/langchain/blob/95dd2f140e19d29bdb62d4dae2048e3edf0ee147/requirements.txt#L12

File: langchain/text_splitter.py (hunk 1 above)
Comment (delip) [reply langchain/text_splitter.py]: done

File: tests/unit_tests/test_text_splitter.py (hunk 2 above)
Comment (delip) [reply tests/unit_tests/test_text_splitter.py]: done

**Comment by delip** [comment]
- [x] make integration_tests

**Review by hwchase17 — APPROVED** [review]

**Pull request merged**